core/src/main/scala/kafka/tools/MirrorMaker.scala

```diff
@@ -83,6 +89,11 @@ object MirrorMaker extends Logging with KafkaMetricsGroup {
   def main(args: Array[String]) {
 
     info("Starting mirror maker")
+    println("WARNING: The default partition assignment strategy of mirror maker will change from 'range' to 'roundrobin' " +
```

Comment:
Before committing to this, I'm wondering what the plan is for compatibility. I doubt users care much whether MM uses range or round-robin, but they would probably expect that they can safely do a rolling upgrade. So a couple points:

1. What does the old consumer do if assignment strategies differ? If we can't make the change in a compatible way, perhaps we should just leave the default as range. We're ultimately planning to deprecate the old consumer anyway.
2. It's actually possible with the new consumer to make this change in a compatible way by enabling both round-robin and range assignment. The coordinator ensures that an assignment strategy which all consumers support is chosen. The downside is that it increases the overhead of the rebalance protocol (basically the subscriptions need to be sent to the coordinator twice). It might make sense to recommend doing two rolling bounces in that case: the first one enables both assignors, the second enables only round-robin. On the other hand, there probably aren't many users who have as many topics as LinkedIn, so perhaps the extra overhead is not significant and it may be reasonable to turn on both assignors by default?